src/features/domain-settings/Steps/Footers/DetailsStepFooter.tsx

```diff
@@ -0,0 +1,116 @@
+import { FC } from 'react';
+
+import { ConfirmActionType, steps } from '../..';
+import { useDomainSettingsData } from '../../hooks';
+
+import { FormErrorText } from '../../../ui';
+import { Button, Step } from '@zero-tech/zui/components';
+import { InfoTooltip } from '@zero-tech/zui/components/InfoTooltip';
+import { IconLock1, IconLockUnlocked1 } from '@zero-tech/zui/components/Icons';
+
+import styles from './Footer.module.scss';
+
+interface DetailsStepFooterProps {
+	zna: string;
+	errorText: string;
+	onSubmit: (action: ConfirmActionType, step: Step, formHeader: string) => void;
+}
+
+export const DetailsStepFooter: FC<DetailsStepFooterProps> = ({
+	zna,
+	errorText,
+	onSubmit,
+}) => {
+	const { truncatedLockedByAddress, isLockedByOwner, metadataLockedStatus } =
+		useDomainSettingsData(zna);
+
+	const tooltipContent = metadataLockedStatus
+		? 'Metadata is locked. Only the person who locked it may unlock and make changes.'
+		: 'You may save changes leaving the metadata unlocked for the next owner to edit, or save & lock the metadata preventing future edits by anyone other than you.';
+
+	return (
+		<>
+			{errorText && (
+				<FormErrorText className={styles.ErrorText} text={errorText} />
+			)}
+
+			{!errorText && metadataLockedStatus && (
+				<WarningLabel
+					isLockedByOwner={isLockedByOwner}
+					truncatedLockedByAddress={truncatedLockedByAddress}
+				/>
+			)}
+
+			<div className={styles.Buttons}>
+				{metadataLockedStatus && isLockedByOwner && (
+					<>
+						<IconLock1 className={styles.LockedIcon} />
+						<Button
+							onPress={() =>
+								onSubmit(ConfirmActionType.UNLOCK, steps[1], 'Unlock Metadata?')
```

Comment:
Couple things that could make this cleaner

1. Wrap function

```js
const onSubmitDetails = (action, formHeader) => {
    return onSubmit(action, steps[1], formHeader)
}
```

2. Set more rigid responsibilities

I think the `DetailsStepFooter` should only be responsible for calling `onSubmit` with an action type, and should leave the current step and form header to the form hook. The `DetailsStepFooter` shouldn't be worried about what the current step is, or what the header text should be.

Ideally, you'd just call `onSubmit(action)`, and the form hook would set internal state for `step` and `formHeader` based on the current step and the action called. Keeping your components as dumb as possible (having as few responsibilities as possible) is a great way to simplify.

I may be missing something in the functionality though so if the above doesn't apply then all good!

Reply:
This should be looking cleaner now, updated mate!